engine/access/rest/accounts_test.go

```diff
@@ -0,0 +1,167 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	mock2 "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/access/mock"
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+const expandableFieldKeys = "keys"
+const expandableFieldContracts = "contracts"
+
+func accountURL(t *testing.T, address string, height string) string {
+	u, err := url.ParseRequestURI(fmt.Sprintf("/v1/accounts/%s", address))
+	require.NoError(t, err)
+	q := u.Query()
+
+	if height != "" {
+		q.Add("block_height", height)
+	}
+
+	u.RawQuery = q.Encode()
+	return u.String()
+}
+
+func TestGetAccount(t *testing.T) {
+	backend := &mock.API{}
+
+	t.Run("get by address at latest sealed block", func(t *testing.T) {
+
+		account := accountFixture(t)
+
+		req := getAccountRequest(t, account, sealedHeightQueryParam, expandableFieldKeys, expandableFieldContracts)
+
+		backend.Mock.
+			On("GetAccountAtLatestBlock", mock2.Anything, account.Address).
+			Return(account, nil)
+
+		expected := expectedExpandedResponse(account)
+
+		assertOKResponse(t, req, expected, backend)
+
```

Comment:
If I understand correctly you think we would need to also assert arguments passed to the mocked grpc methods? The thing is that in the current implementation that anyhow happens, since the arguments provided for the mock method must match the arguments actually passed, else there is a panic that a method was not found.

Reply:
I added `mock.AssertExpectations` for accounts, collections, execution result (where it was fairly easy to add :) )
The one extra thing that `mock.AssertExpectations` checks is the number of calls made to the backend, so I feel its worth checking (atleast where the calls can be easily counted).